src/main/java/io/jenkins/plugins/sprp/models/Agent.java

```diff
@@ -9,9 +11,17 @@
     private String dockerfile;
     private String dir;
     private boolean reuseNode;
+    private String anyOrNone;
 
     Agent(){}
 
+    Agent(String anyOrNone) throws NotSupportedException {
```

Comment:
Use an enum instead of a string. This way no need to check for exception.

Reply:
If the user will type something else other than "none" or "any" we have to throw an exception right?

Reply:
Ah I did not realize `anyOrNone` was coming from user input. I do not know where to check for invalid user input. May this is correct then.